exonum/src/runtime/dispatcher.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use super::{
+    error::{DeployError, ExecutionError, InitError},
+    ArtifactSpec, DeployStatus, CallInfo, EnvContext, InstanceInitData, RuntimeEnvironment,
+};
+
+#[derive(Default)]
+struct DispatcherBuilder {
+    runtimes: HashMap<String, Box<dyn RuntimeEnvironment>>,
+}
+
+impl DispatcherBuilder {
+    pub fn with_runtime(
+        mut self,
+        runtime_name: String,
+        runtime: Box<dyn RuntimeEnvironment>,
+    ) -> Self {
+        self.runtimes.insert(runtime_name, runtime);
+
+        self
+    }
+
+    pub fn finalize(self) -> Dispatcher {
+        Dispatcher {
+            runtimes: self.runtimes,
+        }
+    }
+}
+
+#[derive(Default)]
+struct Dispatcher {
+    runtimes: HashMap<String, Box<dyn RuntimeEnvironment>>,
+}
+
+impl Dispatcher {}
```

Comment:
What does it mean?

Reply:
It means that most probably `Dispatcher` entity will have some methods in the future, but currently it doesn't :)